### ISSUE TITLE
Problem building on Arch linux

### ISSUE THREAD

**Issue by nictaylr** [mono/mono#7200]
## Steps to Reproduce

1. Attempt to build on linux

## Current Behavior

Build fails, with:
```
/usr/bin/ld: cannot find -lpthread-liconv
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:1333: libmonosgen-2.0.la] Error 1
make[4]: *** Waiting for unfinished jobs....
/usr/bin/ld: cannot find -lpthread-liconv
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:1330: libmonoboehm-2.0.la] Error 1
```
etc.

**cannot find -lpthread-liconv**
what does this mean? shouldn't it be `-lpthread` `-liconv`?

## Expected Behavior
Builds properly.

### On which platforms did you notice this

[ ] macOS
[x] Linux
[ ] Windows

**Version Used**:
`git` https://github.com/mono/mono/commit/8ce0775a031af68f1c4db8b738a420ae95986123

**Comment by lewurm** [comment]
have you installed the mentioned libraries, pthread and iconv?

**Comment by nictaylr** [comment]
Yes, I have.

**Comment by lewurm** [comment]
how do you call `autogen.sh`/`configure`? It looks like `LDFLAGS` got messed up a little bit, it should be `-lpthread -liconv` not `-lpthread-liconv` (not the missing space).

**Comment by nictaylr** [comment]
I'm using:
```
./autogen.sh --prefix=/usr  \
    --sysconfdir=/etc   \
    --bindir=/usr/bin   \
    --sbindir=/usr/bin    \
    --disable-quiet-build \
    --with-mcs-docs=no
```

**Comment by nictaylr** [comment]
I just checked, the `LDFLAGS` are still similarly messed up even with just simply `./autogen.sh --prefix=/usr`

**Comment by lewurm** [comment]
can you call make with `V=1`, like so:
```sh
$ make V=1
```
and paste the last couple of lines here. Also `config.log` would be useful for determine the root issue (I think you can attach files by drag & drop here).

**Comment by nictaylr** [comment]
[config.log](https://github.com/mono/mono/files/1749868/config.log)

last few lines from make (everything before is compile commands)
```
/usr/bin/ld: cannot find -lpthread-liconv
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:1333: libmonosgen-2.0.la] Error 1
make[4]: *** Waiting for unfinished jobs....
/usr/bin/ld: cannot find -lpthread-liconv
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:1330: libmonoboehm-2.0.la] Error 1
mv -f .deps/mono_sgen-main-sgen.Tpo .deps/mono_sgen-main-sgen.Po
mv -f .deps/mono_boehm-main.Tpo .deps/mono_boehm-main.Po
make[4]: Leaving directory '/home/nic/Development/build/mono/mono/mini'
make[3]: *** [Makefile:1228: all] Error 2
make[3]: Leaving directory '/home/nic/Development/build/mono/mono/mini'
make[2]: *** [Makefile:484: all-recursive] Error 1
make[2]: Leaving directory '/home/nic/Development/build/mono/mono'
make[1]: *** [Makefile:558: all-recursive] Error 1
make[1]: Leaving directory '/home/nic/Development/build/mono'
make: *** [Makefile:488: all] Error 2
```

**Comment by jaykrell** [comment]
Just missing a space somewhere..

**Comment by nictaylr** [comment]
Ah, great!
Thanks for the help!

**Comment by akoeplinger** [comment]
Fixed by #7230
